plugins/file_chooser/linux/src/file_chooser_plugin.cc

```diff
@@ -159,7 +161,7 @@ void FileChooserPlugin::HandleMethodCall(const MethodCall &method_call,
     result->NotImplemented();
     return;
   }
-  chooser_res = gtk_native_dialog_run(GTK_NATIVE_DIALOG(chooser));
+  chooser_res = gtk_dialog_run(GTK_DIALOG(chooser));
```

Comment:
As a result, I believe on line 181 we'll need to run `gtk_widget_destroy` instead of `g_object_unref`. This appears to be the way the example works, anyway.  Can't remember the specifics around destroying the widget vs un-reffing the object.

[Regular dialog](https://developer.gnome.org/gtk3/stable/GtkFileChooserDialog.html) vs [Native](https://developer.gnome.org/gtk3/stable/gtk3-GtkFileChooserNative.html)

Reply:
Sorry, it‘s my fault, it must be changed to `gtk_widget_destroy`.